src/python/pants/fs/fs_test.py

```diff
@@ -106,3 +109,44 @@ def test_is_child_of(self):
 
     assert not is_child_of(Path("/other/random/directory/root/dist/dir"), mock_build_root)
     assert not is_child_of(Path("../not_root/dist/dir"), mock_build_root)
+
+
+class SingleFileExecutableTest(TestBase):
+
+  def test_raises_with_multiple_files(self):
+    input_files_content = InputFilesContent((
+      FileContent(path='a.txt', content=b'test file contents'),
+      FileContent(path='b.txt', content=b'more test file contents'),
+    ))
+
+    snapshot = self.request_single_product(Snapshot, input_files_content)
+
+    with self.assertRaisesWithMessage(
+        SingleFileExecutable.ValidationError,
+        f'snapshot {snapshot} used for {SingleFileExecutable} should have exactly 1 file!'):
+      SingleFileExecutable(snapshot)
+
+  def test_raises_empty_digest(self):
+    snapshot = Snapshot(EMPTY_DIRECTORY_DIGEST, files=('a.txt',), dirs=())
+
+    with self.assertRaisesWithMessage(
+        SingleFileExecutable.ValidationError,
+        f'snapshot {snapshot} used for {SingleFileExecutable} should have a non-empty digest!'):
+      SingleFileExecutable(snapshot)
+
+  def test_accepts_single_file_snapshot(self):
+    input_files_content = InputFilesContent((
+      FileContent(path='subdir/a.txt', content=b'test file contents'),
+    ))
+    snapshot = self.request_single_product(Snapshot, input_files_content)
+
+    self.assertEquals(SingleFileExecutable(snapshot).exe_filename,
+                      './subdir/a.txt')
```

Comment:
Nit: `assert` preferred because Pytest gives really nice output to show each value was derived.

Reply:
Thank you!! Should I be using this for all testing now?